app/models/pipeline_run.rb

```diff
@@ -276,6 +281,30 @@ def db_load_ercc_counts
     update(total_ercc_reads: ercc_counts_array.map { |entry| entry[:count] }.sum)
   end
 
+  def download_amr_results
+    amr_s3_path = "#{host_filter_output_s3_path}/#{AMR_OUTPUT_NAME}"
+    amr_results = PipelineRun.download_file(amr_s3_path, local_txt_path)
+    amr_results
+  end
+
+  def db_load_amr_counts
+    amr_results = download_amr_results
```

Comment:
I'd rename `amr_results` to indicate that it is a file path.

Also, the fact that the code for downloading this file is spread over 2 different functions makes it a bit confusing to see what's going on. I'd almost just write 
```
amr_results_file = PipelineRun.download_file("#{host_filter_output_s3_path}/#{AMR_OUTPUT_NAME}",
  "#{LOCAL_AMR_PATH}/#{id}")
```
and remove the functions `local_txt_path` and `download_amr_results`. That way it is obvious what we're doing: download the AMR output from S3 to a local folder unique to the pipeline_run id. [If `local_txt_path` were to stay, it should definitely be renamed to something that indicates its purpose, which is to make sure we have a unique folder for each pipeline_run id.]

Even better, write:
```
amr_results_file = PipelineRun.download_file(s3_file_for("amr_counts"), "#{LOCAL_AMR_PATH}/#{id}")
```
so that you don't have to write `"#{host_filter_output_s3_path}/#{AMR_OUTPUT_NAME}"` out twice (again, this makes it more maintainable).

Reply:
Resolved -- from a testing perspective, though, I thought perhaps two functions would be easier? i.e. can test downloading separately that way. But this definitely makes sense from a readability perspective and have made this change!